core/src/main/scala/tamer/Tamer.scala

```diff
@@ -289,8 +298,13 @@ object Tamer {
         sinkRegistry: ULayer[RegistryInfo],
         log: LogWriter[Task]
     ) = for {
-      fiber <- sinkStream(kvStream, sinkRegistry, log)
-      _     <- sourceStream(kvChunkQueue, stateRegistry, log).ensuringFirst(stopSource(log)).ensuring(drainSink(fiber, kvStream, sinkRegistry, log))
+      fiber <- sinkStream(kvStream, serdes.keySerializer, serdes.valueSerializer, log).provideSomeLayer[Clock](sinkRegistry)
+      _ <- sourceStream(kvChunkQueue, stateRegistry, log)
+        .ensuringFirst(stopSource(log))
+        .ensuring(
+          drainSink(fiber, kvStream, serdes.keySerializer, serdes.valueSerializer, log)
+            .provideSomeLayer[Clock](sinkRegistry)
+        )
```

Comment:
I like providing the layer here (it was done in a similar fashion earlier), or ever further down!, but at the moment we have two layers for the same type. One we're now providing here, the other we're passing like a regular parameter (I know, I know!) and we're finally providing it in the `sourceStream` method.

I'm starting to wonder if we shouldn't find a way to tag these types so they are different to the compiler and we just provide them at the edge. Wdyt?

Reply:
Why do they need to be tagged? I don't understand. Is it because we allow using different registries for different topics?

Reply:
Ok, re-reading the code it makes sense. I just didn't realized it worked like that.

Reply:
Among the many options I just used type aliases for this. I initially tried with shapeless but it made the code 5x more complicated, case classes also seemed bulky, I thought about zio subtypes but didn't want to add a new dependency.

Reply:
In hindsight I believe we only need `Has[Registry]` and topic name should just be supplied as a normal parameter to the `serializer` method. Also, `deserializer` should have never required anything more than `Has[Registry]`.
I am happy to do this cleanup myself after this PR is merged.

Reply:
> Also, deserializer should have never required anything more than Has[Registry].
True just keep in mind they need to have the same `R` when passed to create a serde.